Calculator-Advanced-Swift-iOS13/Calculator/Controller/ViewController.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Angela Yu on 10/09/2019.
+//  Copyright © 2019 London App Brewery. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    @IBOutlet weak var displayLabel: UILabel!
+    
+    private var isFinishedTypingNumber: Bool = true
+    private var displayValue: Double {
+        get {
+            guard let number = Double(displayLabel.text!) else {
+                fatalError("Cannot convert display label text to a Double.")
+            }
+            
+            return number
+        }
+        
+        set {
+            displayLabel.text = String(newValue)
+        }
+    }
+    
+    private var calculator = CalculatorLogic()
+    
+    @IBAction func calcButtonPressed(_ sender: UIButton) {
+        isFinishedTypingNumber = true
+        calculator.setNumber(displayValue)
+        
+        if let calcMethod = sender.currentTitle {
```

Comment:
guardでネストを浅くすると良いと思います。

```suggestion
                guard let calcMethod = sender.currentTitle, let result = calculator.calculate(symbol: calcMethod) else { return }
```